templates/antora/template.js

```diff
@@ -0,0 +1,324 @@
+    }
+    return {
+      filename: fileName,
+      content: namespaceTemplate({
+        title: namespace,
+        desc: namespaceDescription
+      })
+    };
+  }).forEach(addPageToArchive);
+
+  archive.saveAs(toPath, function (err) {
+    if (err) throw err;
+  });
+};
+
+function YMLNavToAdoc (navyml) {
+  // Api index page
+  var adoc = '* API Reference\n';
+  var pages = navyml[0].pages
+
+  // generate API namespaces
+  pages.forEach(function (namespace) {
+    // main namespace level navigation (namespace index)
+    adoc += '** ' + namespace.url + '\n';
+    namespace.pages.forEach(function (page) {
+      // namespace level pages
+      adoc += '*** xref:' + AntoraNavBaseDir + page.url + '.adoc' + '[' + page.url+ ']\n';
+    });
+  })
+
+  return adoc;
+};
+
+function getNamespaceFromFullName(fullName) {
+  return fullName.split('.').slice(0, -1).join('.');
+}
+
+function getNamespacesFromTypes(types) {
+  var namespaces = [];
+
+  var namespaces = types.reduce(function (namespaces, type) {
+    var fullName = type.fullName.toLowerCase();
+    var namespace = getNamespaceFromFullName(fullName);
+    return namespace && namespaces.indexOf(namespace) === -1 ? namespaces.concat(namespace) : namespaces;
+  }, []);
+
+  return namespaces;
+}
+
+/**
+ * [getNavFile description]
+ * @return {[type]} [description]
+ */
+function getNavFile(types) {
+  var namespaces = getNamespacesFromTypes(types);
+  var pages = namespaces.map(function (namespace) {
+    var innerPages = types.filter(function (type) {
+      var fullName = type.fullName.toLowerCase();
+      return getNamespaceFromFullName(fullName) === namespace;
+    }).map(function (type) {
+      return { url: type.fullName.toLowerCase() };
+    });
+
+    if (namespace === 'tinymce') {
+      innerPages.unshift({
+        url: 'tinymce.root'
+      });
+    }
+    return {
+      url: namespace,
+      pages: innerPages
+    };
+  });
+
+  return [{
+    url: BASE_PATH,
+    pages: pages
+  }];
+}
+
+/**
+ * [description]
+ * @param  {[type]} data [description]
+ * @return {[type]}      [description]
+ */
+function getMemberPages(root, template, data) {
+  var members = data.getMembers(true);
+  data = data.toJSON()
+  data.datapath = data.type + '_' + data.fullName.replace(/\./g, '_').toLowerCase();
+  data.desc = data.desc.replace(/\n/g, ' ');
+
+  data.constructors = []
+  data.methods = []
+  data.properties = []
+  data.settings = []
+  data.events = []
+  data.keywords = []
+  data.borrows = data.borrows || []
+  data.examples = data.examples || []
+
+  var parents = [data].concat(data.borrows.map(function (parentName) {
+    return root.getTypes().find(function (type) {
+      return type.fullName === parentName
+    }).toJSON()
+  }))
+
+  members.forEach(function (member) {
+    var parentType = member.getParentType();
+
+    member = member.toJSON();
+    data.keywords.push(member.name);
+    member.definedBy = parentType.fullName
```

Comment:
If you pass in { url, fullName } here instead than the url could be lowercase but the fullName be uppercase for presentation. The templates needs to change to obviously.

Reply:
i will give this a go, i forgot about this hack